### DISCUSSION TITLE
Possible new features

### DISCUSSION THREAD

**Discussion by ArturoDent** [ArturoDent/jump-and-select#7] [Ideas]
Let me know your thoughts on these features:

1. Re-enable using regular expressions as a query in a keybinding;
2. Automatically turn off `multiMode` when switch document/tab;
3. Provide a notification pop-up when no match, i.e., no jump/select can be made;
4. Wrap jumps - so, for example, jumping forward would find the next match even if it were at the beginning of the file;
5. Should `reveal` be at the top, center or bottom of the viewport - currently at the bottom; and
6. Should before/afterCharacter do anything for the ^/$/^$ jump/selects - currently ignored.

For other features you can open a new Discussion.  Thank you for your thoughts.

**Comment by rainymatch** [comment]
1. Just wondering. What was the reason to disable RE for the queries? Not sure I would use it anyways. 
2. Will be observing this but sounds logic. Why would I want to stay in the `multiMode` after changing document/tab. However, exiting with enter seems enough to me at the moment.
3. That would be lovely, any kind of feedback is missing. It also seems slightly inconsistent: There is a visual information about `multiMode` being active (and thus also exited)... but the `singleMode`, so to say, does not communicate such information. So if I press the shortcut to jump forward *by mistake*, I will find out just by the "weird behaviour" which will occure afterwards. So what I am thinking now is: why not have a visual feedback for activating the `singleMode` the same way the `multiMode` is announced? (Not sure if any less aggressive color than red is possible for that notification, if so, I would certainly opt for something gray.) Than in case of failure of the jump, I would see that the `singleMode` was exited. That migth be enough. Accompanied with a warning sound would be even better for me, but I guess there's lot of people who would argue against any sound feedback... "Pop-up" sounds to intrusive to me, would not like that.
I also see that in such case I can exit the `singleMode` by pressing `enter`, but `escape` does not work. I guess escape cannot be enabled, rigth? Not so bad. 
4. Can't think of a use case for myself at the moment.
5. Center would be my personal choice. But I think this also depends on the direction. I see now that it is bottom when jumping forward and top when jumping backward. Which feels logic. I guess I would wish for an option 🙂
6. - nothnig at the moment.
